shared/tests/utility/support/TestYamlConversion.cpp

```diff
@@ -0,0 +1,197 @@
+    GIVEN("A yaml node containing a list of data") {
+        YAML::Node test_values = YAML::LoadFile("tests/yamlConversionTestVals.yaml");
+        // error threshold when comparing doubles
+        static constexpr double ERROR_THRESHOLD = 1e-6;
+        // ints
+        WHEN("A vector is declared using integers from the YAML node") {
+            std::vector<int> int_vector = resolve_expression<int>(test_values["ints"]);
+            // Check that the values are good(container size, data type....)
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(test_values["ints"].size() == int_vector.size());
+            }
+            THEN("Values at each index should be equivalent") {
+                // Check values at each index
+                for (size_t i = 0; i < test_values["ints"].size(); i++) {
+                    REQUIRE(test_values["ints"][i].as<int>() == int_vector.at(i));
+                }
+            }
+        }
+
+        // float
+        WHEN("A vector is declared using float values from the YAML node") {
+            std::vector<float> float_vector = resolve_expression<float>(test_values["floats"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(test_values["floats"].size() == float_vector.size());
+            }
+            THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < test_values["floats"].size(); i++) {
+                    REQUIRE(test_values["floats"][i].as<float>() == float_vector.at(i));
+                }
+            }
+        }
+
+        // double
+        WHEN("A vector is declared using doubles from the YAML node") {
+            std::vector<double> double_vector = resolve_expression<double>(test_values["doubles"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(test_values["doubles"].size() == double_vector.size());
+            }
+            THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < test_values["doubles"].size(); i++) {
+                    Approx epsilon_result = Approx(double_vector.at(i)).epsilon(ERROR_THRESHOLD);
+                    REQUIRE(test_values["doubles"][i].as<double>() == epsilon_result);
+                }
+            }
+        }
+
+        // Vector2d
+        WHEN("A vector is declared using Vector2d values from the YAML node") {
+            std::vector<Eigen::Vector2d> v2d_vec = resolve_expression<Eigen::Vector2d>(test_values["vector2"]);
+            THEN("Container sizes should be equivalent") {
+                REQUIRE(test_values["vector2"].size() == v2d_vec.size());
+            }
+            THEN("Values at each index should be equivalent") {
+                for (size_t i = 0; i < test_values["vector2"].size(); i++) {
+                    Approx epsilon_x = Approx(v2d_vec.at(i).x()).epsilon(ERROR_THRESHOLD);
```

Comment:
same here. Checking `Approx` for the floating types should be kept consistent in this file IMO. Do it for all, or for none

Reply:
Roger that.